be/src/olap/like_column_predicate.h

```diff
@@ -45,33 +47,115 @@ class LikeColumnPredicate : public ColumnPredicate {
+                        if (null_map_data[i]) {
+                            if constexpr (is_and) {
+                                flags[i] &= _opposite;
+                            } else {
+                                flags[i] = _opposite;
+                            }
+                            continue;
+                        }
+
+                        StringValue cell_value = nested_col_ptr->get_value(data_array[i]);
+                        if constexpr (is_and) {
+                            unsigned char flag = 0;
+                            (_state->function)(
+                                    const_cast<vectorized::LikeSearchState*>(&_like_state),
+                                    cell_value, pattern, &flag);
+                            flags[i] &= _opposite ^ flag;
+                        } else {
+                            unsigned char flag = 0;
+                            (_state->function)(
+                                    const_cast<vectorized::LikeSearchState*>(&_like_state),
+                                    cell_value, pattern, &flag);
+                            flags[i] = _opposite ^ flag;
+                        }
+                    }
+                } else {
+                    LOG(FATAL) << "vectorized (not) like predicates should be dict column";
+                }
+            } else {
+                if (column.is_column_dictionary()) {
+                    auto* nested_col_ptr = vectorized::check_and_get_column<
+                            vectorized::ColumnDictionary<vectorized::Int32>>(column);
+                    auto& data_array = nested_col_ptr->get_data();
+                    for (uint16_t i = 0; i < size; i++) {
+                        StringValue cell_value = nested_col_ptr->get_value(data_array[i]);
+                        if constexpr (is_and) {
+                            unsigned char flag = 0;
+                            (_state->function)(
+                                    const_cast<vectorized::LikeSearchState*>(&_like_state),
+                                    cell_value, pattern, &flag);
+                            flags[i] &= _opposite ^ flag;
+                        } else {
+                            unsigned char flag = 0;
+                            (_state->function)(
+                                    const_cast<vectorized::LikeSearchState*>(&_like_state),
+                                    cell_value, pattern, &flag);
+                            flags[i] = _opposite ^ flag;
+                        }
+                    }
+                } else {
+                    LOG(FATAL) << "vectorized (not) like predicates should be dict column";
+                }
+            }
+        }
+    }
+
     std::string _origin;
     // life time controlled by scan node
     doris_udf::FunctionContext* _fn_ctx;
-    doris_udf::StringVal pattern;
+    using PatternType = std::conditional_t<is_vectorized, StringValue, StringVal>;
```

Comment:
Confusing for the type StringValue and StringVal, especially for the starter.  It's better to keep one of them.

Reply:
This confusion will be resolved after we remove the non-vectorized code.

Reply:
Got it